src/pkcs11/pkcs11-global.c

```diff
@@ -590,7 +590,12 @@ CK_RV C_GetSlotInfo(CK_SLOT_ID slotID, CK_SLOT_INFO_PTR pInfo)
 		}
 	}
 
-	if (rv == CKR_TOKEN_NOT_PRESENT || rv == CKR_TOKEN_NOT_RECOGNIZED)
+	if (rv == CKR_TOKEN_NOT_PRESENT || rv == CKR_TOKEN_NOT_RECOGNIZED
+			/* The sub-routines are loosing the context of "C_GetSlotInfo" and
+			 * return a different default value for SC_ERROR_INVALID_CARD than
+			 * CKR_TOKEN_NOT_RECOGNIZED.  Explicitly check for that default
+			 * value, here */
+			|| rv == sc_to_cryptoki_error(SC_ERROR_INVALID_CARD, NULL))
```

Comment:
should not this have the second argument `"C_GetSlotInfo"` so it can get the context?

Reply:
In any case, what you wrote is equivalent to `rv == CKR_FUNCTION_NOT_SUPPORTED`, isn't it?

Reply:
yes, it's intended to translate the default value for SC_ERROR_INVALID_CARD (currently CKR_FUNCTION_NOT_SUPPORTED), because SC_ERROR_INVALID_CARD should not return an error in C_GetSlotInfo.